### PR TITLE
LPD-37783 patchWishList API can add the same item more than once to the same wish list

### PR DESCRIPTION
**Pull request by ghost**
resend this 
https://github.com/brianchandotcom/liferay-portal/pull/156882

@alessiorendina

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:stop

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-wish-list-service/src/main/java/com/liferay/commerce/wish/list/internal/model/listener/GroupModelListener.java [hunk 1]
```diff
@@ -21,10 +25,22 @@ public class GroupModelListener extends BaseModelListener<Group> {
 
 	@Override
 	public void onBeforeRemove(Group group) {
-		_commerceWishListLocalService.deleteCommerceWishListsByGroupId(
-			group.getGroupId());
+		try {
+			CommerceWishListThreadLocal.setDefaultWishListDeletable(true);
+			_commerceWishListLocalService.deleteCommerceWishListsByGroupId(
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-wish-list-service/src/main/java/com/liferay/commerce/wish/list/internal/model/listener/GroupModelListener.java:30]: Add an empty line before

File: modules/apps/commerce/commerce-wish-list-service/src/main/java/com/liferay/commerce/wish/list/internal/model/listener/UserModelListener.java [hunk 2]
```diff
@@ -21,10 +25,22 @@ public class UserModelListener extends BaseModelListener<User> {
 
 	@Override
 	public void onBeforeRemove(User user) {
-		_commerceWishListLocalService.deleteCommerceWishListsByUserId(
-			user.getUserId());
+		try {
+			CommerceWishListThreadLocal.setDefaultWishListDeletable(true);
+			_commerceWishListLocalService.deleteCommerceWishListsByUserId(
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-wish-list-service/src/main/java/com/liferay/commerce/wish/list/internal/model/listener/UserModelListener.java:31]: Same here

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/jaxrs/exception/mapper/RequiredCommerceWishListExceptionMapper.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+@Provider
+public class RequiredCommerceWishListExceptionMapper
+	extends BaseExceptionMapper<RequiredCommerceWishListException> {
+
+	@Override
+	protected Problem getProblem(
+		RequiredCommerceWishListException requiredCommerceWishListException) {
+
+		return new Problem(
+			Response.Status.BAD_REQUEST, "Unable to delete default wishlist");
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/jaxrs/exception/mapper/RequiredCommerceWishListExceptionMapper.java:38]: `wishlist` -> `wish list`

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@CrescenzoRegaLF please check my comments, thank you!

File: modules/apps/commerce/commerce-wish-list-service/src/main/java/com/liferay/commerce/wish/list/service/impl/CommerceWishListItemServiceImpl.java [hunk 4]
```diff
@@ -159,6 +174,26 @@ public int getCommerceWishListItemsCount(long commerceWishListId)
 			commerceWishListId);
 	}
 
+	@Override
+	public CommerceWishListItem updateCommerceWishListItem(
+		long commerceAccountId, long commerceWishListId, String cpInstanceUuid,
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-wish-list-service/src/main/java/com/liferay/commerce/wish/list/service/impl/CommerceWishListItemServiceImpl.java:179]: Please indent correctly

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:report:92261115

**Comment by CrescenzoRegaLF** [comment]
ci:report:92263391

**Comment by alessiorendina** [comment]
@CrescenzoRegaLF confirmed that failures are not related to this PR, thank you!

**Pull request closed without merging**